admin/components/swlogin.js

```diff
@@ -0,0 +1,39 @@
+(function (staticWeb) {
+    "use strict";
+    var Login = function () {
+        if (!(this instanceof Login)) {
+            return new Login();
+        }
+
+        return this.init();
+    }
+    Login.prototype = {
+        createInterface: function () {
+          staticWeb.includeStyle(staticWeb.getAdminPath() + 'css/swadminzone.css');
+          var templateName = 'swlogin-LoggedOut';
+          if(staticWeb.hasLoggedInInfo()) {
+            templateName = 'swlogin-LoggingIn';
+          }
+          staticWeb.retrieveTemplate(templateName, function(template) {
+            var elements = staticWeb.elements.swlogin.instances;
```

Comment:
should this be `staticWeb.retrieveInstanceElements('swlogin')` instead? So that we may introduce or modify javascript object structure? The way we're using the object is like it's a public API of the core platform.